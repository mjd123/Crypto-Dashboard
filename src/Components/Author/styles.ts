import styled from "styled-components";
import { PageText as AuthorText } from "../../helpers/PageText";
import { Wrapper as AuthorWrapper } from "../../helpers/Wrapper";
import { device } from "../../styles/Breakpoints";

export const Text = styled(AuthorText)`
  color: ${(props) => (props.color ? props.color : "#ffffff")};
  font-size: ${(props) => (props.fontSize ? props.fontSize : ".9em")};
`;

export const Wrapper = styled(AuthorWrapper)`
  display: flex;
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.alignItems};
  padding: 20px;
  grid-row: 3;
  grid-column: 1;
  ${device.sm`
    padding:5px;
  `};
`;
