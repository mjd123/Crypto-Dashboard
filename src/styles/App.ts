import styled from "styled-components";
import { device } from "./Breakpoints";

interface Props {
  backgroundImage: string | undefined;
}

export const App = styled.main<Props>`
  width: 100vw;
  height: 100vh;
  padding: 10px;
  background-image: url("
    ${(Props) => (Props.backgroundImage ? Props.backgroundImage : "")} ");
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  display: grid;
  grid-template-columns: 50% 50%;
  grid-template-rows: auto auto auto;
  box-shadow: inset 0 0 0 100vmax rgba(0, 0, 0, 0.2);
  overflow: hidden;
  overflow-y: auto;
  position: fixed;

  ${device.sm`
   grid-template-columns: 100%;

  `};
`;
