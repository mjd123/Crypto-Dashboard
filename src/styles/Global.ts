import { createGlobalStyle } from "styled-components";
import { device } from "./Breakpoints";

export default createGlobalStyle`
   *{
       margin: 0;
       padding: 0;
       outline:0;
       box-sizing:border-box;
       font-family: 'Open Sans', sans-serif; 
       box-sizing: border-box;
   }
   #root{
       margin:0 auto;
       box-sizing: border-box;
       height: 200vh;
   }

   body {
        @media only screen and (min-width: 768px) {
            overflow: hidden;
        }
   }
`;
