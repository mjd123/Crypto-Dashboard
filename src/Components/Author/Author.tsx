import { Text, Wrapper } from "./styles";

interface AuthorProps {
  name?: string | undefined;
}

const Author = ({ name }: AuthorProps) => {
  return (
    <Wrapper
      className="author-wrapper"
      justifyContent="start"
      alignItems="flex-end"
    >
      {name && <Text>Photo by: {name}</Text>}
    </Wrapper>
  );
};

export default Author;
