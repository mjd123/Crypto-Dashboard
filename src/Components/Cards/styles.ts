import styled from "styled-components/macro";
import { PageText as CardText } from "../../helpers/PageText";
import { Wrapper as CardWrapper } from "../../helpers/Wrapper";
import { device } from "../../styles/Breakpoints";

export const Text = styled(CardText)`
  color: ${(props) => (props.color ? props.color : "#ffffff")};
  font-size: ${(props) => (props.fontSize ? props.fontSize : ".9em")};
  white-space: nowrap;
`;

export const Wrapper = styled(CardWrapper)`
  //display: flex;
  position: relative;
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.justifyContent};
  padding: 10px;
  grid-row: 2 / 4;
  grid-column: 2;
  box-shadow: inset 0 0 0 100vmax rgba(0, 0, 0, 0.6);
  border-radius: 25px;
  margin: 20px;
  justify-self: end;
  width: fit-content;
  height: fit-content;
  min-height: 300px;
  min-width: 300px;
  .ant-card :not(.expanded) {
    display: grid;
    grid-template: 1fr /35px auto;
    align-items: center;
    justify-content: space-evenly;
    text-align: center;
    padding: 10px;

    .ant-card-body {
      padding: 0;
      display: grid;
      grid-template: 1fr /0 1fr 1fr 1fr;
      align-items: center;
      grid-column: 2;
      grid-row: 1;
      width: 100%;
    }
  }

  ${device.md`
    grid-column: 1 / span 2;
    margin:auto;
    justify-self:center;
    height:fit-content;
  `};
`;

export const CardTextWrapper = styled(CardWrapper)`
  display: ${(props) => props.display};
  flex-direction: ${(props) => props.flexDirection};
  justify-content: ${(props) => props.justifyContent};
  height: ${(props) => (props.height ? props.height : "100%")};
  width: 100%;
  grid-row: ${(props) => props.gridRow};
  margin-bottom: ${(props) => props.marginBottom};
  margin-left: ${(props) => props.marginLeft};
  margin-right: ${(props) => props.marginRight};
`;

export const Image = styled.img`
  max-width: 15px;
`;
