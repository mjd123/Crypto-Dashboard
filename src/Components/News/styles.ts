import styled from "styled-components";
import { Row } from "antd";
import { PageText as NewsText } from "../../helpers/PageText";
import { Wrapper as NewsWrapper } from "../../helpers/Wrapper";
import { device } from "../../styles/Breakpoints";

export const Text = styled(NewsText)`
  color: ${(props) => (props.color ? props.color : "#ffffff")};
  font-size: ${(props) => (props.fontSize ? props.fontSize : ".9em")};
`;

export const Wrapper = styled(NewsWrapper)`
  position: relative;
  grid-row: auto;
  grid-column: 1;
  box-shadow: inset 0 0 0 100vmax rgba(0, 0, 0, 0.6);
  border-radius: 25px;
  margin: 20px;
  width: 100%;
  max-width: 300px;
  height: auto;
  padding: 10px;
  justify-self: start;
  width: fit-content;
  ${device.md`
    grid-column: 1 / span 2;
    margin:auto;
    justify-self:center;
    height:fit-content;
  `};
`;

export const StyledRow = styled(Row)`
  height: auto;
  margin: 0;
`;
