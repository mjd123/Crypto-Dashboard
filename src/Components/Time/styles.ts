import styled from "styled-components";
import { PageText as TimeText } from "../../helpers/PageText";
import { Wrapper as TimeWrapper } from "../../helpers/Wrapper";
import { device } from "../../styles/Breakpoints";

export const Text = styled(TimeText)`
  color: ${(props) => (props.color ? props.color : "#ffffff")};
  font-size: ${(props) => (props.fontSize ? props.fontSize : ".9em")};
`;

export const Wrapper = styled(TimeWrapper)`
  display: flex;
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.justifyContent};
  padding: 20px;
  ${device.sm`
    justify-content:center;
    padding:5px;
    align-self:auto;
  `};
`;
