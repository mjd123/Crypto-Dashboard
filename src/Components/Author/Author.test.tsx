import React from "react";
import { shallow, mount } from "enzyme";
import Author from "./Author";
import { Wrapper, Text } from "./styles";
import toJson from "enzyme-to-json";

test("Author renders time with given string", () => {
  const mockProps = {
    name: "string",
  };
  const component = mount(<Author name={mockProps.name} />);
  const text = component.find("span").text();

  expect(typeof text).toBe("string");
  expect(text).toEqual("Photo by: string");
});
