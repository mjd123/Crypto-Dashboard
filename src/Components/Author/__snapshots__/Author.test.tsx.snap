// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `ShallowWrapper {}`;

exports[`Author renders time with given string 1`] = `"string"`;
